borrow-collateral-ratio/src/agent.spec.js

```diff
@@ -0,0 +1,158 @@
+const mockContract = {};
+
+jest.mock('forta-agent', () => ({
+  ...jest.requireActual('forta-agent'),
+  getEthersProvider: jest.fn(),
+  ethers: {
+    ...jest.requireActual('ethers'),
+    Contract: jest.fn().mockReturnValue(mockContract),
+  },
+}));
+
+const {
+  Finding, FindingType, FindingSeverity, ethers,
+} = require('forta-agent');
+
+const { provideInitialize, provideHandleBlock } = require('./agent');
+const { getAbi } = require('./utils');
+const config = require('../bot-config.json');
+
+// check the configuration file to verify the values
+describe('check bot configuration file', () => {
+  it('protocolName key required', () => {
+    const { protocolName } = config;
+    expect(typeof (protocolName)).toBe('string');
+    expect(protocolName).not.toBe('');
+  });
+
+  it('protocolAbbreviation key required', () => {
+    const { protocolAbbreviation } = config;
+    expect(typeof (protocolAbbreviation)).toBe('string');
+    expect(protocolAbbreviation).not.toBe('');
+  });
+
+  it('developerAbbreviation key required', () => {
+    const { developerAbbreviation } = config;
+    expect(typeof (developerAbbreviation)).toBe('string');
+    expect(developerAbbreviation).not.toBe('');
+  });
+
+  it('contracts key required', () => {
+    const { contracts } = config;
+    expect(typeof (contracts)).toBe('object');
+    expect(contracts).not.toBe({});
+  });
+
+  it('contracts key values must be valid', () => {
+    const { contracts } = config;
+    Object.keys(contracts).forEach((key) => {
+      const { address, abiFile } = contracts[key];
+
+      expect(address).not.toBe(undefined);
+      // check that the address is a valid address
+      expect(ethers.utils.isHexString(address, 20)).toBe(true);
+
+      // load the ABI from the specified file
+      // the call to getAbi will fail if the file does not exist
+      getAbi(abiFile);
+    });
+  });
+
+  it('ratioSettings values must be valid', () => {
+    const { ratioSettings } = config;
+    const { maxUtilizationRate, type, severity } = ratioSettings;
+
+    // check that maxUtilizationRate is of type 'number'
+    expect(typeof (maxUtilizationRate)).toBe('number');
+
+    // check type, this will fail if 'type' is not valid
+    expect(Object.prototype.hasOwnProperty.call(FindingType, type)).toBe(true);
+
+    // check severity, this will fail if 'severity' is not valid
+    expect(Object.prototype.hasOwnProperty.call(FindingSeverity, severity)).toBe(true);
+  });
+});
+
+describe('monitor borrow to collateral ratio', () => {
+  describe('handleBlock', () => {
+    let initializeData;
+    let handleBlock;
+    const mockSymbol = 'TEST';
+    const mockAssetTokenAddress = '0xMOCKASSETTOKEN';
+
+    beforeEach(async () => {
+      initializeData = {};
+
+      const mockReturnValue = [mockSymbol, mockAssetTokenAddress];
+      mockReturnValue.symbol = mockSymbol;
+      mockReturnValue.tokenAddress = mockAssetTokenAddress;
```

Comment:
This is interesting.  Is it due to how ethers.js will parse out the return value?

Or, will the following suggestion work?
```suggestion
      const mockReturnValue = {
        symbol: mockSymbol,
        tokenAddress: mockAssetTokenAddress
      };
```

Reply:
Yep, this is in response to how ethers.js returns parsed log values. ethers.js uses an array so that values can be placed in order of the parameters of the corresponding call. Additionally, ethers.js makes a best effort to provide named key/values for each parameter - this is possible to add to an array since an array in js is also just an object. Returning an object explicitly here would technically work, but since ethers.js will return an array I chose to more closely resemble the output that ethers.js would provide.